Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -1564,20 +1559,19 @@ - (void)setDisappearingMessagesConfiguration:
     [self updateBarButtonItems];
 }
 
-- (void)updateMessageMappingRangeOptions:(MessagesRangeSizeMode)mode
+- (void)updateMessageMappingRangeOptions
 {
-    // The "old" range length may have been increased by insertions of new messages
-    // at the bottom of the window.
-    NSUInteger oldLength = [self.messageMappings numberOfItemsInGroup:self.thread.uniqueId];
-
-    NSUInteger targetLength = oldLength;
-    if (mode == MessagesRangeSizeMode_Truncate) {
+    NSUInteger rangeLength = 0;
+    
+    if (self.lastRangeLength == 0) {
```

Comment:
We were using the "mode" to special case the first time we calculate the range size.  Now we can just use `lastRangeLength`.  We can now execute this logic more than once (for empty conversations) but that's fine.